### ISSUE TITLE
'Reportlab import canvas' issue

### ISSUE THREAD

**Issue by ikwtif** [pyinstaller/pyinstaller#2083]
Hi. I'm trying to convert my script to an exe but having some issues. The *.exe just closes. When I run it in cmd I get the error below. I just started learning python so my knowledge is very limited so I have no idea what I'm supposed to do now, some help would be appreciated. I found a post of someone else with the same issue but his fix did not work for me when I tried it: http://stackoverflow.com/questions/35691320/issues-with-pyinstaller-and-reportlab so it seems to be an issue with reportlab. In "my save_a_pdf_2.py" on line 2 I have this code "from reportlab.pdfgen import canvas" wich seems to be causing the issue from what I can gather. However, I have no idea how to solve this problem myself.

```
 File "app.py", line 9, in <module>
File "<frozen importlib._bootstrap>", line 969, in _find_and_load
File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
File "c:\users\...\appdata\local\programs\python\python35-32\lib\site-packa
ges\PyInstaller\loader\pyimod03_importers.py", line 389, in load_module
    exec(bytecode, module.__dict__)
File "save_a_pdf_2.py", line 2, in <module>
File "<frozen importlib._bootstrap>", line 969, in _find_and_load
File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
File "c:\users\...\appdata\local\programs\python\python35-32\lib\site-packa
ges\PyInstaller\loader\pyimod03_importers.py", line 389, in load_module
    exec(bytecode, module.__dict__)
File "site-packages\reportlab\pdfgen\canvas.py", line 19, in <module>
File "<frozen importlib._bootstrap>", line 969, in _find_and_load
File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
File "<frozen importlib._bootstrap>", line 634, in _load_backward_compatible
File "c:\users\...\appdata\local\programs\python\python35-32\lib\site-packa
ges\PyInstaller\loader\pyimod03_importers.py", line 389, in load_module
    exec(bytecode, module.__dict__)
File "site-packages\reportlab\rl_config.py", line 131, in <module>
File "site-packages\reportlab\rl_config.py", line 102, in _startUp
File "site-packages\reportlab\lib\utils.py", line 695, in rl_isdir
AttributeError: 'FrozenImporter' object has no attribute '_files'
Failed to execute script app  
```

**Comment by stephenrauch** [comment]
The wiki has a [nice page on how to report bugs](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs) so that the devs have the best chance of helping you.

You did not leave a test script to try and duplicate your problem.  I did however use the minimal example from the SO page you referenced, but that worked fine for me given these versions:

    1369 INFO: PyInstaller: 3.3.dev0+g7b0f45f.mod
    1369 INFO: Python: 2.7.11
    1369 INFO: Platform: Windows-7-6.1.7601-SP1

**Comment by ikwtif** [comment]
Don't have too much time today to do extensive testing, but this is what I came up with for now. Not sure if this is that usefull? If I can do some more specific testing that would be helpful let me know. 

I'm using Python 3.5.1 (v3.5.1:37a07cee5969, Dec  6 2015, 01:38:48) [MSC v.1900 32 bit (Intel)] on win32. With pyinstaller 3.2, wich I installed with the pip installer.
platform: windows 7 ultimate SP1 64bit

I was trying to trackdown the bug, but it seems I might have more than one isse.

- _test code(test1)_


    from tkinter import *
    from tkinter import ttk
    root = Tk()
    content = ttk.Frame(root)
    button = ttk.Button(content)
    def save_pdf():
        pass
    mbutton_save = ttk.Button(text ='save pdf', command=lambda:  save_pdf())
    mbutton_save.grid(row= 0,column= 0)`

wich gives me this warning log: https://www.dropbox.com/s/ua57gqm12tdd9pd/warnapp.txt?dl=0

cmd log of build: "Error loading Python DLL: C:\Users\Dieter\AppData\Local\Programs\Python\Python35
-32\Scripts\build\app\python35.dll (error code 126)"
cmd log of dist: nothing


- _code with the original save_pdf import(test2)_
cmd log of build: "Error loading Python DLL: C:\Users\Dieter\AppData\Local\Programs\Python\Python35
-32\Scripts\build\app\python35.dll (error code 126)"
cmd log of dist: https://www.dropbox.com/s/gvxuxgxcwe71hbx/cmd%20dist%20log.txt?dl=0
warning log: https://www.dropbox.com/s/b4i6vfk6qiy4v8o/warnapp.txt?dl=0


- _full original code(test main)_
cmd log of build: "Error loading Python DLL: C:\Users\Dieter\AppData\Local\Programs\Python\Python35
-32\Scripts\build\app\python35.dll (error code 126)"
cmd log of dist: https://www.dropbox.com/s/altly8sa98rcwc2/cmd%20dist%20log.txt?dl=0
warning log: https://www.dropbox.com/s/cz1idio8mfyxyao/warnapp.txt?dl=0

dropbox link with the app scripts and dist/build + logs: https://www.dropbox.com/sh/a6z5n1oclxl7vze/AACOOYkSklbemApNX84hKNSsa?dl=0

**Comment by htgoebel** [comment]
@ikwtif Please post the requested logs in the issue, not on an external service like dropbox. I will not access it for privacy reasons.

**Comment by htgoebel** [comment]
Beside of my last comment:

According to the code lines shown in the stackoverflow-thread, this is a bug in ReportLab. This lines shows that ReportLab assumes the loader to have a private attribute `_files`. Looking at the [code](https://bitbucket.org/rptlab/reportlab/src/7f3365bf11f2/src/reportlab/lib/utils.py) ReportLab tries to figure out whether it is packed by some tool like McMillan Installer (which is the base of PyInstaller).

Please [submit a bug-report for ReportLib](https://bitbucket.org/rptlab/reportlab/issues).

**Comment by anonaut** [comment]
this issue is fixed in reportlab 3.3.19
see https://bitbucket.org/rptlab/reportlab/issues/94/using-reportlab-with-pyinstaller-fails
